### PR TITLE
APP-1339 - Changes to block the main buttons while executing an action

### PR DESCRIPTION
**Pull request by luanapp**

### CODE REVIEW

File: src/main/java/org/symphonyoss/integration/jira/api/JiraApiResource.java [hunk 1]
```diff
@@ -121,6 +126,30 @@ public ResponseEntity searchAssignableUsers(@RequestParam String issueKey,
   }
 
   /**
+   * Get information of a specific issue.
+   * @param issueKey Issue identifier
+   * @return List of potential assigneers users or 400 Bad Request - Returned if no issue key
```
Comment (robson-sanchez) [inline src/main/java/org/symphonyoss/integration/jira/api/JiraApiResource.java]: It seems this comment was copy and paste from another method.

File: src/main/webapp/templates/dialog.hbs [hunk 2]
```diff
@@ -1,19 +1,31 @@
...
         <div class="header">
+            {{#if headerError}}
+            <div class="headerError"><img src="../img/exclamation_mark.svg" />{{headerError}}</div>
+            {{/if}}
             <div class="label">
                 {{actionText}} <a href="{{url}}">{{key}}</a>
             </div>
             <h1>{{{subject}}}</h1>
+            <div class="label key-value">
+                <span class="key">Assignee</span>
```
Comment (robson-sanchez) [inline src/main/webapp/templates/dialog.hbs]: According to the proposed design, this key should be 'Current Assignee'.

File: src/main/webapp/templates/dialog.hbs [hunk 3]
```diff
@@ -1,19 +1,31 @@
...
+                <span class="value">{{assignee}}</span>
+            </div>
         </div>
         <div class="content">
             <alert isVisible="{{showError}}">{{errorMessage}}</alert>
```
Comment (rsanchez-daitan) [inline src/main/webapp/templates/dialog.hbs]: I believe this alert is not required anymore.

File: src/main/webapp/templates/builders/dialogBuilder.js [hunk 4]
```diff
@@ -10,6 +10,11 @@ export default class DialogBuilder {
     this.innerContent = innerContent;
     this.showFooter = true;
     this.showError = false;
```
Comment (rsanchez-daitan) [inline src/main/webapp/templates/builders/dialogBuilder.js]: It seems not required anymore.

File: src/main/webapp/templates/builders/dialogBuilder.js [hunk 5]
```diff
@@ -10,6 +10,11 @@ export default class DialogBuilder {
...
+    this.isLoading = false;
+  }
+
+  headerError(errorMessage) {
+    this.headerError = errorMessage;
   }
 
   error(message) {
```
Comment (rsanchez-daitan) [inline src/main/webapp/templates/builders/dialogBuilder.js]: The same comment above.

File: src/main/webapp/services/assignUserService.js [hunk 6]
```diff
@@ -69,25 +72,43 @@ export default class AssignUserService extends BaseService {
...
-      const dialogBuilder = new DialogBuilder('Assign', assignTemplate);
+    if (this.selectedUser.email === undefined) {
       dialogBuilder.error('Please select an user');
 
       const template = this.retrieveTemplate(dialogBuilder, data, this.serviceName);
       this.updateDialog('assignIssue', template.layout, template.data);
     } else {
+      dialogBuilder.loading(true);
+
+      const template = this.retrieveTemplate(dialogBuilder, data, this.serviceName, true);
```
Comment (robson-sanchez) [inline src/main/webapp/services/assignUserService.js]: I think we should change the button label to 'SAVING' when the attribute loading equals true.

**Review by robson-sanchez — CHANGES_REQUESTED** [review]
I think you should apply the behavior to block action button in the comment dialog as well.

**Comment by robson-sanchez** [comment]
@luanapp Please verify the conflicting files.

File: src/main/webapp/services/assignUserService.js (hunk 6 above)
Comment (luanapp) [reply src/main/webapp/services/assignUserService.js]: Fixed!

File: src/main/java/org/symphonyoss/integration/jira/api/JiraApiResource.java (hunk 1 above)
Comment (luanapp) [reply src/main/java/org/symphonyoss/integration/jira/api/JiraApiResource.java]: Fixed!

File: src/main/webapp/templates/dialog.hbs (hunk 2 above)
Comment (luanapp) [reply src/main/webapp/templates/dialog.hbs]: Fixed!

**Review by robson-sanchez — APPROVED** [review]

**Pull request merged**